ignite/metrics/precision.py

```diff
@@ -15,7 +15,7 @@ class _BasePrecisionRecall(_BaseClassification):
     def __init__(
         self,
         output_transform: Callable = lambda x: x,
-        average: bool = False,
+        average: Union[bool, str] = False,
```

Comment:
For the first point, I've done it in `Precision` and `Recall` themselves because their set of `average` options are not the same. I didn't get the second point. You mean if we got `False`, set `average` to `'false'` to be like other options?

Reply:
You are right we are missing `None` value as scikit-learn which would match previous `average=False`. We should also deprecate `is_multilabel` as it becomes unused. What do you think ?

> I've done it in Precision and Recall themselves because their set of average options are not the same.

Can you detail a bit more this point ? I haven't checked but IMO they should be same.

Reply:
Yes, we could remove `multilabel`. It could be infered from data.
`Recall` has not the `weighted` option, but regarding the next comment, I'll add it so they become the same.

Reply:
I was thinking that `average="samples"` is equivalent to `is_multilabel=True, average=True`. Can we use `average="samples"` for binary case ?

Reply:
`average='samples'` for binary case is equivalent with the term below which is not a known metric:
![image](https://user-images.githubusercontent.com/22097587/169614438-692fce18-66f5-42dd-a26d-1aa476588030.png)
According to Scikit, `average='samples'` is "only meaningful for multilabel classification".
Anyway, without `is_multilabel`, we cannot differentiate binary data from multilabel data because we do not have a differentiator assumption on their dimensions. Scikit can, because it assumes binary data is 1-dimensional and multilabel data is 2-dimensional.

Reply:
I may be misunderstand your point, but I was thinking about new API like that:
```python
pr = Precision(average="samples")  # old: Precision(is_multilabel=True, average=True)
pr.update(multilabel_data)

pr = Precision()  # old: Precision()
pr.update(binary_data)

pr = Precision(average="samples")  # old: Precision(is_multilabel=True, average=True)
pr.update(binary_data)  # try to interpret binary_data as multilabel
```

> Anyway, without is_multilabel, we cannot differentiate binary data from multilabel data because we do not have a differentiator assumption on their dimensions. Scikit can, because it assumes binary data is 1-dimensional and multilabel data is 2-dimensional.

By the way, scikit learn has "binary" option as well. If we stuck with that, we can add it as well...

Reply:
Do you still think we can remove `is_multilabel`?
About `average='binary'`, Scikit uses it only for binary data. When average is binary, it returns the precision for class `pos_label` which is another parameter. So if `pos_label` is 1 (default), the simple precision is returned, and if the `pos_label` is 0, `TN/(TN+FN)` (precision for class 0) is returned.
I think we don't need this option. If user wants precision for class 0, he/she can simply do `1-y` and `1-y_pred` in the `output_transform`.
In our current implementation, when average is `True`, we return the simple precision (precision for class 1) for binary data.

Reply:
> Do you still think we can remove is_multilabel?

We do not remove it but just deprecate for removing later. For me if average can take str values, e.g. "samples", `is_multilabel` is redundant and can be deprecated. What do you think ?

Reply:
If we depracate `is_multilabel`, how could we differentiate binary from multilabel in the case average is anything but 'samples'?

Reply:
If average is not 'samples' then input data is not multilabel: binary or multiclass.

Reply:
Other options of average are also compatible with multilabel data, so if average is not 'samples', intput data can still be multilabel.

Reply:
> Other options of average are also compatible with multilabel data, so if average is not 'samples', intput data can still be multilabel.

Can you give a concrete example ? Even if it is so, is output still meaningful and can be compared to scikit learn ?

Reply:
The formulas I just added to docstrings are for multi label as well. As an example, suppose we have exam result of some students and  exam questions are YES/NO questions. Exam results is multilabel data. If we measure precision considering questions altogether, it would be micro precision. If we calculate  precision for each question then measure (weighted) average of them, it would be (weighted) macro precision.

Yes. I adopted weighted, samples and micro from Scikit, and basically used it in tests as well.